### ISSUE TITLE
Install fails on python 3.11: Couldn't load plugin 'profiletool' due to an error when calling its classFactory() method 

### ISSUE THREAD

**Issue by rpfigueira** [PANOimagen/profiletool#84]
The installation of the Profile tool plugin in QGIS 3.26 installed on a Fedora 37 machine with python 3.11 fails with the  following error:
```
Couldn't load plugin 'profiletool' due to an error when calling its classFactory() method 

re.error: global flags not at the start of the expression at position 62 
Traceback (most recent call last):
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 423, in _startPlugin
    plugins[packageName] = package.classFactory(iface)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/__init__.py", line 28, in classFactory
    from .profileplugin import ProfilePlugin
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/profileplugin.py", line 40, in 
    from .tools.profiletool_core import ProfileToolCore
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/tools/profiletool_core.py", line 48, in 
    from ..ui.ptdockwidget import PTDockWidget
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/ui/ptdockwidget.py", line 40, in 
    from ..tools.plottingtool import *
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/tools/plottingtool.py", line 44, in 
    from .. import pyqtgraph as pg
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/pyqtgraph/__init__.py", line 202, in 
    from .graphicsItems.VTickGroup import *
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/pyqtgraph/graphicsItems/VTickGroup.py", line 7, in 
    from .. import functions as fn
  File "/usr/lib64/python3.11/site-packages/qgis/utils.py", line 888, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/pyqtgraph/functions.py", line 41, in 
    FLOAT_REGEX = re.compile(r'(?P[+-]?((((\d+(\.\d*)?)|(\d*\.\d+))([eE][+-]?\d+)?)|((?i)(nan)|(inf))))\s*((?P[u' + SI_PREFIXES + r']?)(?P\w.*))?$')
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/__init__.py", line 227, in compile
    return _compile(pattern, flags)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/__init__.py", line 294, in _compile
    p = _compiler.compile(pattern, flags)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_compiler.py", line 743, in compile
    p = _parser.parse(p, flags)
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 980, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 455, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 863, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 455, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 863, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 455, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 863, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 455, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/re/_parser.py", line 841, in _parse
    raise source.error('global flags not at the start '
re.error: global flags not at the start of the expression at position 62


Python version: 3.11.2 (main, Feb  8 2023, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)] 
QGIS version: 3.26.1-Buenos Aires Buenos Aires, exported 
```

**Comment by nicogodet** [comment]
I can't really upgrade pyqtgraph as it cause issues on old platforms (see https://github.com/PANOimagen/profiletool/issues/62).
You can upgrade pyqtgraph locally by manually replacing it on plugin folder.

**Comment by JaviBecerra** [comment]
Hi @rpfigueira,

Thanks for reporting. As @nicogodet already noted, it is an issue with the embedded version of pyqtgraph shipped with profiletool, and updating pyqtgraph has already caused some issues in the past :/

It seems however that the only problem with the pyqtgraph version that we are shipping (re python 3.11) lies on two regular expressions in file `profiletool/pyqtgraph/functions.py`.

As a workaround, you can update the definition of FLOAT_REGEX and INT_REGEX in your installation ("/home/<user>/.local/share/QGIS/QGIS3/profiles/demo/python/plugins/profiletool/pyqtgraph/functions.py) with those below:

```python
FLOAT_REGEX = re.compile(r'(?P<number>[+-]?((((\d+(\.\d*)?)|(\d*\.\d+))([eE][+-]?\d+)?)|((?i:nan)|(inf))))\s*((?P<siPrefix>[u' + SI_PREFIXES + r']?)(?P<suffix>\w.*))?$')
INT_REGEX = re.compile(r'(?P<number>[+-]?\d+)\s*(?P<siPrefix>[u' + SI_PREFIXES + r']?)(?P<suffix>.*)$')
```
We have made some tests and this fix seems to work fine with newer python 3.11 without breaking previous versions, let us know if you find some issue anyway.

Regards,
Javier

**Comment by rpfigueira** [comment]
Thank you both for the quick reply! I tested the solution by @JaviBecerra, which solved the issue.

**Comment by nicogodet** [comment]
Fixed in [654c055](https://github.com/PANOimagen/profiletool/commit/654c055442425a2be5845a85a84780f5397f7259)

Release 4.2.5 should be available to install as soon as plugin update has been approved.
